src/Spatial/Euclidean/StraightLine2D.cs

```diff
@@ -0,0 +1,315 @@
+        /// </summary>
+        /// <param name="points">The points though which the line should pass</param>
+        /// <exception cref="ArgumentException">Thrown when less than 2 points are provided</exception>
+        /// <exception cref="ArgumentException">Thrown when points are not collinear</exception>
+        public StraightLine2D(IEnumerable<Point2D> points) : this(points.ToArray())
+        {
+        }
+
+        /// <summary>
+        /// Creates a Line conforming to the equation y = mx + b which passes through the points provided
+        /// </summary>
+        /// <param name="points">The points though which the line should pass</param>
+        /// <exception cref="ArgumentException">Thrown when less than 2 points are provided</exception>
+        /// <exception cref="ArgumentException">Thrown when points are not collinear</exception>
+        public StraightLine2D(Point2D[] points)
+        {
+            if (points.Length < 2)
+            {
+                throw new ArgumentException("A line must be constructed from at least two points");
+            }
+
+            Gradient = (points[1].Y - points[0].Y) / (points[1].X - points[0].X);
+            if (double.IsInfinity(Gradient))
+            {
+                yoffset = double.NaN; // special case vertical line has no y-intercept
+                xoffset = points[1].X;
+            }
+            else
+            {
+                yoffset = points[1].Y - (Gradient * points[1].X);
+                xoffset = double.NaN;
+            }
+
+            for (int i = 2; i < points.Length; i++)
+            {
+                if (!IsThrough(points[i]))
+                {
+                    throw new ArgumentException("The points provided are not collinear");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Determines if the line is horizontal
+        /// </summary>
+        /// <returns>True if the line is horizontal</returns>
+        public bool IsHorizontal()
+        {
+            return Gradient == 0;
+        }
+
+        /// <summary>
+        /// Determines if the line is vertical
+        /// </summary>
+        /// <returns>returns true if the line is vertical</returns>
+        public bool IsVertical()
+        {
+            return double.IsInfinity(Gradient);
+        }
+
+        /// <summary>
+        /// Tests if the other line is parallel to this one
+        /// </summary>
+        /// <returns>Returns true if lines are parallel</returns>
+        public bool IsParallel(ILine otherLine)
+        {
+            if (otherLine is StraightLine2D)
+            {
+                IsParallel((StraightLine2D)otherLine);
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Tests if the other line is parallel to this one
+        /// </summary>
+        /// <returns>Returns true if lines are parallel</returns>
+        public bool IsParallel(StraightLine2D otherLine)
```

Comment:
Should probably have a tolerance parameter, not sure what a good default is. An overload with `Angle tolerance` can probably make sense.

Reply:
Currently the internal implementation uses doubles to store the gradients but it is preferable that longer term that this stores the line internally as an equation using Mathnet.symbolics.  This would avoid the need for a tolerance and I think it would be better not to add to the API now only to remove it later

Reply:
In other places we have named it `IsParallelTo` https://github.com/mathnet/mathnet-spatial/blob/master/src/Spatial/Euclidean/Line2D.cs#L182

Reply:
Who's the end user of the library? Since not everyone can afford symbolic computations, I assume (for example) that game developers aren't part of the user base.